### ISSUE TITLE
Live activity 

### ISSUE THREAD

**Issue by dugamarian** [gui-dos/DiaBLE-Legacy#22]
Hi,

The app seems to be great, but I noticed that it doesn’t have notifications. In my opinion, live activities  would be a great feature in order to show BG. 

Thanks!

**Comment by gui-dos** [comment]
Yep, this is of course in my TODOs also on the new main active repo I am actively working again on: https://github.com/gui-dos/DiaBLE/blob/main/TODO.md .

I was sharing with Glucose Direct most of the low level code base and decided to focus on exploring the internals of the Libre 3 while Reimar was getting popular on TestFlight and Facebook with the new final user solutions... :-)

Unfortunately all the other iOS projects also like xDrip4iO5 and LibreTrasmitter seem dead in the water regarding the Libre 3: I hope to keep my spirit high and keep on playing with the latest Swift idioms and SwiftUI classes.

**Comment by dugamarian** [comment]
Is a Great app and I think it would make the app much more intuitive. I use GlucoseDirect, but libre2 has become a CGM, meaning it behaves like libre3 and sends readings minute by minute to LibreLinkUp server. Even though it works connected directly, it seems that the readings of the LibreLink app are more accurate. That's why I thought that using Diable, with the connection to LibreLinkUp, is much more suitable. 
Also, GlucoseDirect doesn’t seem to be maintained anymore.

**Comment by gui-dos** [comment]
My LibreLinnkUp class is more general indeed: I already tested it also with Libre 1s and Libre 2s (even wearing multiple sensors at the same time) and I am providing a (F)ollower mode letting access the data for a `patientId` different than the account logging it.

Are you aware of the latest experimental branch of xDrip4iOS5 https://github.com/paulplant/xdripswift/tree/librelink-follower ? It is already quite stable and Paul Plant can also count on thousand beta testers and a dozen if TestFlight releasers ;-)

**Comment by dugamarian** [comment]
xdrip4ios was the first application I used and I still use it, back when I was using libre1 with blucon. I believe it was the first app that was truly more useful than LibreLink. Thanks for the suggestion.

I am, however, trying to implement live activity in Diable, but by profession, I am a biologist with some experience in R, so it's going to be challenging for me, but I'll keep trying :))

**Comment by gui-dos** [comment]
My code base is four years old and would need a complete rewrite of its foundations.

There have been some quirks but luckily Apple just released the version 4.4 of the Playgrounds for Catalyst/iOS 17and I succeeded in restoring mine (in the past I was supporting also the Blucon and even the Watlaa watch connecting directly with the MiaoMiao and the Bubble).

I am adopting now the new `@Observable` and `#Preview` macros and I am starting to use `Task { @MainActor` instead of  `DispatchQueue.main.async`.

I like to stay on the edge of the latest betas: the next step will be probably the new tracking and scrolling features of the SwiftUI Charts but the major hurdle will be to understand whether it will be ever possible to reverse engineer the native methods to decrypt the Libre 3 data since the current Android solutions like Juggluco still have to leverage on the original obfuscated  libraries which is not technically possible on iOS of course.

I am already having fun however exploring the internal `trident.realm` database which is unveiling many more  technical details: luckily I was able to import the Realm Swift package at the first try on Xcode 15! B-)

Eagerly waiting for your first PRs!
